Makefile

```diff
@@ -49,21 +49,31 @@ PAIRING_ASM_SOURCES = $(wildcard $(ARCHDIR)/*.s)
 CPP_SOURCES = $(PAIRING_CPP_SOURCES)
 ASM_SOURCES = $(PAIRING_ASM_SOURCES)
 
+BIN_OS   ?= $(shell uname)
+BIN_ARCH ?= $(shell uname -m)
+
 BINDIR = bin
 PAIRING_OBJECTS = $(addprefix $(BINDIR)/,$(CPP_SOURCES:.cpp=.o)) $(addprefix $(BINDIR)/,$(ASM_SOURCES:.s=.o))
 
-all: pairing.a
+all: pairing.a ;
+
+$(BINDIR)/builtfor-%:
```

Comment:
Do we need `BIN_OS` and `BIN_ARCH`? They seem like yet another pair of environment variables to manage when the thing that really matters is the CXX, AR, and AS that we're using to build.